pkg/kubelet/volume/cache/actual_state_of_world.go

```diff
@@ -0,0 +1,586 @@
+// PodExistsInVolume() failed to find volume in the list of attached volumes.
+func IsVolumeNotAttachedError(err error) bool {
+	_, ok := err.(volumeNotAttachedError)
+	return ok
+}
+
+// IsRemountRequiredError returns true if the specified error indicates that
+// PodExistsInVolume() found volume/pod attached/mounted but
+// remountRequired was true, indicating the given volume should be remounted to
+// the pod to reflect changes in the referencing pod.
+func IsRemountRequiredError(err error) bool {
+	_, ok := err.(remountRequiredError)
+	return ok
+}
+
+type actualStateOfWorld struct {
+	// hostName is the name of this node. This value is passed to Attach/Detach
+	hostName string
+	// attachedVolumes is a map containing the set of volumes the kubelet volume
+	// manager believes to be successfully attached to this node. Volume types
+	// that do not implement an attacher interface are assumed to be in this
+	// state on detection.
+	// The key in this map is the name of the volume and the value is an object
+	// containing more information about the attached volume.
+	attachedVolumes map[api.UniqueVolumeName]attachedVolume
+	// volumePluginMgr is the volume plugin manager used to create volume
+	// plugin objects.
+	volumePluginMgr *volume.VolumePluginMgr
+	sync.RWMutex
+}
+
+// The volume object represents a volume the kubelet volume manager believes to
+// be successfully attached to a node it is managing. Volume types that do not
+// implement an attacher are assumed to be in this state on detection.
+type attachedVolume struct {
+	// volumeName contains the unique identifier for this volume.
+	volumeName api.UniqueVolumeName
+
+	// mountedPods is a map containing the set of pods that this volume has been
+	// successfully mounted to. The key in this map is the name of the pod and
+	// the value is a mountedPod object containing more information about the
+	// pod.
+	mountedPods map[volumetypes.UniquePodName]mountedPod
+
+	// spec is the volume spec containing the specification for this volume.
+	// Used to generate the volume plugin object, and passed to plugin methods.
+	// In particular, the Unmount method uses spec.Name() as the volumeSpecName
+	// in the mount path:
+	// /var/lib/kubelet/pods/{podUID}/volumes/{escapeQualifiedPluginName}/{volumeSpecName}/
+	spec *volume.Spec
+
+	// pluginName is the Unescaped Qualified" name of the volume plugin used to
+	// attach and mount this volume. It is stored seperatley in case the full
+	// volume spec (everything except the name) can not be reconstructed for a
+	// volume that should be unmounted (which would be the case for a mount path
+	// read from disk without a full volume spec).
+	pluginName string
+
+	// pluginIsAttachable indicates the volume plugin used to attach and mount
+	// this volume implements the volume.Attacher interface
+	pluginIsAttachable bool
+
+	// globallyMounted indicates that the volume is mounted to the underlying
```

Comment:
How about 'globallyMounted indicates that the volume is mount to a global mount point' -- i think NFS follows this rule, and it doesn't use device files afk.